src/main/scala/org/broadinstitute/dsde/firecloud/utils/EnabledUserDirectives.scala

```diff
@@ -0,0 +1,79 @@
+package org.broadinstitute.dsde.firecloud.utils
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.{StatusCode, StatusCodes}
+import akka.http.scaladsl.server.Directives.extractUri
+import akka.http.scaladsl.server.{RequestContext, Route, RouteResult}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.model.UserInfo
+import org.broadinstitute.dsde.firecloud.{FireCloudConfig, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.rawls.model.{ErrorReport, ErrorReportSource}
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi
+import org.broadinstitute.dsde.workbench.client.sam.{ApiClient, ApiException}
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+trait EnabledUserDirectives extends LazyLogging  with SprayJsonSupport {
+
+  /**
+    * Queries Sam to see if the current user is enabled. If the user is disabled,
+    * responds with Forbidden and prevents the rest of the route from executing.
+    *
+    * @param userInfo credentials for the current user
+    * @param samBaseUrl where to find Sam - used for unit testing
+    * @return n/a
+    */
+  def requireEnabledUser(userInfo: UserInfo, samBaseUrl: String = FireCloudConfig.Sam.baseUrl)(innerRoute: RequestContext => Future[RouteResult]): Route = {
+    extractUri { uri =>
+      Try(getUserEnabled(userInfo.accessToken.token, samBaseUrl)) match {
```

Comment:
I think this should probably wrap the blocking Sam call in a Future instead of a Try. Or alternatively use the async client (see [example](https://github.com/broadinstitute/rawls/pull/1946/files#r930090541)).